docs/source/markdown/guides/how_to/data/custom_data.md

```diff
@@ -0,0 +1,289 @@
+commands:
+
+```python
+print(train_data["image"].shape)
+# torch.Size([32, 3, 256, 256])
+
+print(train_data["mask"].shape)
+# torch.Size([32, 256, 256])
+```
+
+Training the model is as simple as running the following command:
+
+```{literalinclude} ../../../../snippets/train/api/segmentation/model_and_engine.txt
+:language: python
+```
+
+where we train a Patchcore model on this custom dataset with default model parameters.
+:::
+
+:::{tab-item} CLI
+:sync: label-2
+Here is the CLI command to create the same custom datamodule:
+
+```{literalinclude} ../../../../snippets/config/data/image/folder/segmentation/cli/default.yaml
+:language: yaml
+```
+
+Assume that we have saved the above config file as `segmentation.yaml`. We could run the following CLI command to train a Patchcore model on above dataset:
+
+```bash
+anomalib train --data segmentation.yaml --model anomalib.models.Patchcore
+```
+
+:::
+
+::::
+:::::
+
+This example demonstrates how to create a segmentation dataset with
+normal and abnormal images. We could expand this example to create a
+segmentation dataset with only normal images.
+
+### With Only Normal Images
+
+There are certain cases where we only have normal images in our dataset
+but would like to train a segmentation model. Anomalib provides
+synthetic anomaly generation capabilities to create abnormal images from
+normal images. We could use the [Folder](../../reference/data/image/folder.md) datamodule to train a model on
+this dataset.
+
+:::::{dropdown} Code Syntax
+:icon: code
+
+::::{tab-set}
+:::{tab-item} API
+:sync: label-1
+
+We could run the following python code to create the custom datamodule:
+
+```{literalinclude} ../../../../snippets/data/image/folder/segmentation/normal.txt
+:language: python
+```
+
+As can be seen from the code above, we only need to specify the
+`test_split_mode` argument to `SYNTHETIC`. The [Folder](../../reference/data/image/folder.md) datamodule will create training, validation, test and prediction datasets and
+dataloaders for us.
+
+To check what individual samples from dataloaders look like, we can run
+the following command:
+
+```{literalinclude} ../../../../snippets/data/image/folder/segmentation/dataloader_values.txt
+:language: python
+```
+
+We could check the shape of the images and masks using the following
+commands:
+
+```python
+print(train_data["image"].shape)
+# torch.Size([32, 3, 256, 256])
+
+print(train_data["mask"].shape)
+# torch.Size([32, 256, 256])
+```
+
+:::
+
+:::{tab-item} CLI
+:sync: label-2
+
+Here is the CLI command to create the same custom datamodule with only normal images. We only need to change the `test_split_mode` argument to `SYNTHETIC` to generate synthetic anomalies.
+
+```{literalinclude} ../../../../snippets/config/data/image/folder/segmentation/cli/normal.yaml
+:language: yaml
+```
+
+Assume that we have saved the above config file as `normal.yaml`. We could run the following CLI command to train a Patchcore model on above dataset:
```

Comment:
should we mention that the parameters can also be entered directly as CLI arguments, instead of using a `.yaml` file?

Reply:
I think this would be too much of a detail for this tutorial. It would be good to show all these details in the CLI tutorial.